src/main.rs

```diff
@@ -208,15 +208,18 @@ fn handle_input_events(
         Err((Some(ENODEV), _)) => Ok(false),
         Err((_, error)) => Err(error).context("Error fetching input events"),
         Ok(events) => {
+            let mut vec: Vec<InputEvent> = Vec::new();
             for event in events {
                 if event.event_type() == EventType::KEY {
                     handler
                         .on_event(event, config)
                         .map_err(|e| anyhow!("Failed handling {event:?}:\n  {e:?}"))?;
                 } else {
-                    handler.send_event(event)?;
+                    vec.push(event);
```

Comment:
This means that, if you interleave key and non-key events, they could be reordered. Is that intentional? I feel like you should push multiple events at once only for the ranges that don't include the other type of events.

Reply:
True I didn't think of the reordering.
Shkuld I send them in bulk only when absolute input is detected?